### PR TITLE
[DEVOPS-4852] Optional pod ports

### PR DESCRIPTION
**Pull request by anaharris**
Jira: https://codecademy.atlassian.net/browse/DEVOPS-4852

Main changes:
- make pod port optional, by adding logic to service deployment and removing the default value for it
- add and update test CI values files, so that they test scenarios with and without `containerPort` value
- version bump

### CODE REVIEW

File: charts/service/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: service
 description: A generic k8s service chart
 type: application
-version: 1.1.4
+version: 1.2.0
```
Comment (anaharris) [inline charts/service/Chart.yaml:5]: i'm not aware of all consequences of this change, and therefor version update that this PR should introduce. are any services relying on port that was defined in `charts/service/values.yaml`?  those serve as default values, right?

Comment (lindsaylandry) [reply charts/service/Chart.yaml:5]: other apps will refer to this chart like so:
https://github.com/codecademy-engineering/Codecademy/blob/develop/.helm/codecademy/requirements.yaml#L7

Comment (anaharris) [reply charts/service/Chart.yaml:5]: got it! but just to confirm, removing [this line](https://github.com/codecademy-engineering/helm-charts/blob/main/charts/service/values.yaml#L51) won't break anything?

Comment (anaharris) [reply charts/service/Chart.yaml:5]: for context, we talked and v1.2.0 makes more sense since we're removing default value

**Pull request merged**